### PR TITLE
1.2/develop

### PR DESCRIPTION
**Pull request by steadweb**
Refine 400 status codes by throwing 'correct' exception(s). 

At the moment, all 400 status codes are thrown as \RequestStatusException. If environment is set to development, all 400 statues bar 404 throw \Exception. This gives the developer more information when debugging, and not just a "Not Found" page!

### CODE REVIEW

**Comment by WanWizard** [comment]
@philsturgeon any comments?

File: classes/request/curl.php [hunk 1]
```diff
@@ -179,7 +179,27 @@ public function execute(array $additional_params = array())
 		elseif ($this->response->status >= 400)
 		{
 			$this->set_defaults();
-			throw new \RequestStatusException($body, $this->response->status);
+						
+			// Only show specific 400 error message with \Exception if not in production mode
+			if(Fuel::$env != 'production')
```
Comment (Daniel15) [inline classes/request/curl.php]: This should use the Fuel::PRODUCTION constant.

Comment (steadweb) [inline classes/request/curl.php]: Thanks Dan. I'll amend now.

**Comment by frankdejonge** [comment]
I'm throwing my two cents in. While I understand where this is coming from, this can lead to unexpected behavior. When in production mode the server could also throw a non 404 status. For example when there is work being done on that server. I think the way to go about this, is to fetch the exception, check the status code, and re-throw (or handle differently) based on that, not based on what kind of exception is thrown.

**Comment by WanWizard** [comment]
I agree that handling the exception is application specific, and should be dealt with there.

Closing this request.

**Pull request closed without merging**